akka-stream-tests/src/test/scala/akka/stream/impl/StreamLayoutSpec.scala

```diff
@@ -168,6 +174,33 @@ class StreamLayoutSpec extends StreamSpec {
         fut.futureValue(veryPatient) should ===(List(42))
       }
 
+      "starting from a future Source" in {
+        val g = Source.fromFutureGraph(Future {
+          Thread.sleep(2000)
+          Fusing.aggressive((1 to tooDeepForStack).
+            foldLeft(Source.single(42).mapMaterializedValue(_ ⇒ 1))(
+              (f, i) ⇒ f.map(identity)))
+        })
+
+        val (mat, fut) = g.toMat(Sink.seq)(Keep.both).run()
+        mat.futureValue(veryPatient) should ===(1)
+        fut.futureValue(veryPatient) should ===(List(42))
+      }
+
+      "starting from a completion stage of Source" in {
+        val future: Future[Graph[SourceShape[Int], Int]] = Future {
+          Fusing.aggressive((1 to tooDeepForStack).
+            foldLeft(Source.single(43).mapMaterializedValue(_ ⇒ 1))(
+              (f, i) ⇒ f.map(identity)))
+        }
+        val stage: CompletionStage[Graph[SourceShape[Int], Int]] = future.toJava
+        val g = Source.fromGraphCompletionStage(stage)
+
+        val (mat, fut) = g.toMat(Sink.seq)(Keep.both).run()
+        mat.toScala.futureValue(veryPatient) should ===(1)
+        fut.futureValue(veryPatient) should ===(List(43))
+      }
+
```

Comment:
Why does these go in the `StreamLayoutSpec`, they when they cover the `FutureFlattenSource` and not any changes to the `StreamLayout`?